jax/_src/random.py

```diff
@@ -926,21 +961,39 @@ def _next_kxv(kxv):
     return key, X, V, U
 
   # initial state is chosen such that _cond_fn will return True
+  key, subkey = _split(key)
+  u_boost = uniform(subkey, (), dtype=dtype)
   _, _, V, _ = lax.while_loop(_cond_fn, _body_fn, (key, zero, one, _lax_const(alpha, 2)))
-  z = lax.mul(lax.mul(d, V), boost)
-  return lax.select(lax.eq(z, zero), jnp.finfo(z.dtype).tiny, z)
+  if log_space:
+    # TODO(jakevdp): there are negative infinities here due to issues mentioned above. How should
+    # we handle those?
```

Comment:
how about setting lower bound (through minval) for u_boost? `0` value is not expected for the algorithm I believe. It is even better than using `1 - u` (which has lower bound eps, rather than tiny)

Reply:
That's a good idea... what we should really do is base this on `exponential()`, and make those kinds of decisions there. But we can't because then the log samples would not be numerically equivalent to the non-log samples, and we can't change the non-log samples without breaking backward compatibilty for the sampler. So I think until we're comfortable breaking users, the algorithm I have here is the only option.